### ISSUE TITLE
[New UI] End-of-line comments break highlighting

### ISSUE THREAD

**Issue by AdamantConlanger** [def-gthill/lexurgy#86]
TL;DR: In the new UI, this minimal example renders with no syntax highlighting at all on the second line.
Minimal example:
```
Feature +voiced #
Feature +long
```

Details: If I use the new UI, the syntax highlighting seems to work as intended in almost all cases. Weirdly, though, comments seem to break it in some very specific edge cases.
Specifically, a comment preceded by at least one space on a non-empty feature-declaring or symbol-declaring line causes the remainder of the document to evade all syntax highlighting. This doesn't happen if the comment isn't preceded by a space, nor if it occupies its own line, nor if it's placed at the end of a different type of declaration, name, or rule.

**Comment by def-gthill** [comment]
The new UI [now has its own public repo](https://github.com/def-gthill/lexurgy-app). Please open this and any future new UI issues there.
